### ISSUE TITLE
Plugin is not sending DNS updates "socket.gaierror: [Errno -8]"

### ISSUE THREAD

**Issue by gdtoro** [sjm-steffann/netbox-ddns#33]
**Hello, 
I get the following errors when I try to update or create a DNS record:** 

>  Traceback (most recent call last):
>    File "/opt/netbox/venv/lib/python3.10/site-packages/rq/worker.py", line 1359, in perform_job
>      rv = job.perform()
>    File "/opt/netbox/venv/lib/python3.10/site-packages/rq/job.py", line 1178, in perform
>      self._result = self._execute()
>    File "/opt/netbox/venv/lib/python3.10/site-packages/rq/job.py", line 1215, in _execute
>      result = self.func(*self.args, **self.kwargs)
>    File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/background_tasks.py", line 189, in dns_delete
>      delete_forward(dns_name, address, status, output)
>    File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/background_tasks.py", line 83, in delete_forward
>      response = dns.query.udp(update, zone.server.address, port=zone.server.server_port)
>    File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/models.py", line 116, in address
>      addrinfo = socket.getaddrinfo(self.server, 'domain', proto=socket.IPPROTO_UDP)
>    File "/usr/lib/python3.10/socket.py", line 955, in getaddrinfo
>      for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
>  **socket.gaierror: [Errno -8] Servname not supported for ai_socktype**

**Could you please help me with that issue? 
Netbox v3.4.4**

**Comment by gdtoro** [comment]
Yes, of course. 
![tempadwawds](https://user-images.githubusercontent.com/76843498/224631263-4915d7aa-e46b-48ca-9383-0e34d2101c73.JPG)

![image](https://user-images.githubusercontent.com/76843498/224631493-d3d60a2e-c983-4133-9023-b26cea3a196c.png)

**Comment by YWatchman** [comment]
Ok, alright. So the code explicitly searches for a name, not an IP Address as the code converts this internally to an IP Address. You can get around this by adding a hostfile entry or just creating DNS entry for the DNS primary.

In a next release, I'll check if the input value seems an IPAddress of some sorts and do not do the conversion. :-)

**Comment by gdtoro** [comment]
There doesn't seem to be any difference when you specify the IP address or DNS name. Resault is the same. 
```
netbox-worker_1           File "/opt/netbox/venv/lib/python3.10/site-packages/rq/job.py", line 1178, in perform
netbox-worker_1             self._result = self._execute()
netbox-worker_1           File "/opt/netbox/venv/lib/python3.10/site-packages/rq/job.py", line 1215, in _execute
netbox-worker_1             result = self.func(*self.args, **self.kwargs)
netbox-worker_1           File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/background_tasks.py", line 170, in dns_create
netbox-worker_1             create_forward(dns_name, address, status, output)
netbox-worker_1           File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/background_tasks.py", line 50, in create_forward
netbox-worker_1             response = dns.query.udp(update, zone.server.address, port=zone.server.server_port)
netbox-worker_1           File "/opt/netbox/venv/lib/python3.10/site-packages/netbox_ddns/models.py", line 116, in address
netbox-worker_1             addrinfo = socket.getaddrinfo(self.server, 'domain', proto=socket.IPPROTO_UDP)
netbox-worker_1           File "/usr/lib/python3.10/socket.py", line 955, in getaddrinfo
netbox-worker_1            for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
netbox-worker_1         socket.gaierror: [Errno -8] Servname not supported for ai_socktype
```

![image](https://user-images.githubusercontent.com/76843498/225070578-afc29186-7278-4d1e-8c69-f0883b3e0ad5.png)
![image](https://user-images.githubusercontent.com/76843498/225070909-b2d8aeb0-3f9d-44dc-b7d6-71b5567dbcd3.png)

Also I forgot the mention that this is a nebox docker instalation.

**Comment by zejar** [comment]
I stumbled across the same issue with my NetBox 3.5 setup and created pull-request #35 to pass the correct information to the `socket.getaddrinfo` function.

**Comment by YWatchman** [comment]
Fixed by #35
